### PR TITLE
 Expanding on the documentation for callable classes

### PR DESCRIPTION
**Pull request by MahmoudShawkyy**
I have added explanation and some examples with showing expected errors when dealing with callable classes

### CODE REVIEW

**Comment by parlough** [comment]
Thanks for opening a PR @MahmoudShawkyy! Actually, I think the previous example and surrounding text was already good for what's covered here for a few reasons:

1. We want the text to read more as a reference than a tutorial that gets right to the point.
2. I don't think we need to call out examples with and without parameters. We want the section to be pretty focused since the language tour is long already, so I think having the first example showcase this should be sufficient since the normal function documentation covers all the functionality functions support. If we want to be more explicit in how it supports return types, parameters, etc, we should just say that and link to the the "Functions" section where they can learn more about that functionality. Especially since this a more advance topic and readers likely are already familiar with functions.
3. The same goes for the added note, since the same limitation applies to all functions in Dart.
4. Also before, this was a runnable DartPad which allowed users to run it and see what was happening. That's super useful and not something we want to lose.

In https://github.com/dart-lang/site-www/issues/4267, there are three main goals I was thinking:

1. Make the implicit tear-off behavior clear, that is discussed more here (https://github.com/dart-lang/site-www/issues/2071).
2. Document what use cases this is useful/important in, rather than just what it does. 
3. Perhaps more explicitly link to the [Functions](https://dart.dev/guides/language/language-tour#functions) section with an indication that the same functionality is supported.

Perhaps point 3 is the most important to handle and should cover the changes you made here. It could look something like this with just some slight modifications from the original text:

```
To allow an instance of your Dart class to be called like a function,
implement the `call()` method.

The `call()` method is a special method
that allows any class that defines ones to emulate a function,
while supporting the same functionality as normal [functions](#functions)
such as parameters and return types.

In the following example, the `WannabeFunction` class defines a `call()` function
that takes three strings and concatenates them, separating each with a space,
and appending an exclamation. Click **Run** to execute the code.
```

What do you think about keeping the current combined example and just modifying the text like above to make it more explicit how it supports that other functionality? Thanks again!

**Comment by MahmoudShawkyy** [comment]
Thanks @parlough I have reviewed the text and examples and I have added your modified suggested code to the the callable classes section

File: src/_guides/language/language-tour.md [hunk 1]
```diff
@@ -4544,13 +4544,17 @@ Iterable<int> naturalsDownFrom(int n) sync* {
...
 ## Callable classes
 
 To allow an instance of your Dart class to be called like a function,
 implement the `call()` method.
 
-In the following example, the `WannabeFunction` class defines a call() function
+The `call()` method is a special method
+that allows any class that defines it to emulate a function,
+while supporting the same functionality as normal [functions](#functions)
+such as parameters and return types.
```
Comment (atsansone) [inline src/_guides/language/language-tour.md:4555]: **issue:** Tighten the language. The `...is a special method that...` does not aid in reader understanding. I would also cut this into shorter sentences.

```suggestion
The `call()` method allows any class that defines it to emulate a function.
This method supports the same functionality as normal [functions](#functions)
such as parameters and return types.
```

**Review by atsansone — DISMISSED** [review]
@MahmoudShawkyy : I would tighten up the language some in this change.

**Review by parlough — APPROVED** [review]
Thanks @MahmoudShawkyy! I committed the suggested changes and this looks good to me.

**Pull request merged**